src/libraries/System.Composition.TypedParts/src/System/Composition/TypedParts/Discovery/DiscoveredPart.cs

```diff
@@ -130,6 +135,34 @@ private IEnumerable<CompositionDependency> GetPartActivatorDependencies(Dependen
             }
         }
 
+        private ConstructorInfo GetConstructorInfoFromGenericType(TypeInfo type)
+        {
+            var genericPartTypeInfo = type.GetGenericTypeDefinition().GetTypeInfo();
```

Comment:
Why call GetGenericTypeDefinition twice?

```suggestion
            var genericPartTypeInfo = genericPartType .GetTypeInfo();
```